src/dmd/expression.d

```diff
@@ -3177,15 +3177,13 @@ extern (C++) final class StructLiteralExp : Expression
              *   __sl%s%d, where %s is the struct name
              */
             const size_t len = 10;
-            char[len] buf = void;
+            char[len + 1] buf = void;
+            buf[len] = 0;
+            strcpy(buf.ptr, "__sl");
+            strncat(buf.ptr, sd.ident.toChars(), len - 4 - 1);
```

Comment:
strncat must die. Nobody remembers if the terminating 0 is put there or not, and if it is there, is it after the length or not. 99% of the time when I see a strncat, it's being incorrectly used.